app/helpers/application_helper.rb

```diff
@@ -364,4 +364,17 @@ def get_user_access_token(user)
   def pluralize_without_count(count, noun, text=nil)
     count.to_i == 1 ? "#{noun}#{text}" : "#{noun.pluralize}#{text}"
   end
+
+  # get the rails route specification in a format suitable
+  # for use in analytics or logging
+  # e.g. /single_cell/study/:accession/:study_name
+  def get_route_spec_string
+    spec_name = 'undefined'
```

Comment:
@bistline this is waaaay grosser than I thought it would have to be to get the string from routes.rb.  I spent about half an hour digging through SO and apiDock before settling on this--I'm hoping you know an easier way.  I had assumed there'd be an easy method for this named something like  `current_route.match_spec`